src/material/core/theming/_theming.scss

```diff
@@ -432,9 +432,10 @@ $_internals: _mat-theming-internals-do-not-access;
 // Checks for duplicate styles in a `theme-version: 0` style theme.
 @mixin _check-duplicate-theme-styles-v0($theme-or-color-config, $id) {
   $theme: private-legacy-get-theme($theme-or-color-config);
-  $color-config: get-color-config($theme);
-  $density-config: get-density-config($theme);
-  $typography-config: get-typography-config($theme);
+  $color-config: map.get($theme, $_internals, m2-config, color) or get-color-config($theme);
```

Comment:
you could expose this function internally if you want: https://github.com/angular/components/blob/main/src/material/core/theming/_m2-inspection.scss#L44, would probably make the logic here slightly simpler. Though this is fine too